### ISSUE TITLE
examples with "%" in names fail

### ISSUE THREAD

**Issue by joshgav** [terraform-providers/terraform-provider-azurerm#494]
In the example for [storage_share](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/website/docs/r/storage_share.html.markdown#example-usage) and [storage_table](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/website/docs/r/storage_table.html.markdown) a `%d` and `%s` are included in the names. The provider rejects this as an invalid name.

To address this I submitted #493, are there better options?

### Terraform Version
0.10.8

### Affected Resource(s)
Please list the resources as a list, for example:
- storage_share
- storage_table

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "test" {
  name     = "acctestrg-%d"
  location = "westus"
}

resource "azurerm_storage_account" "test" {
  name                     = "acctestacc%s"
  resource_group_name      = "${azurerm_resource_group.test.name}"
  location                 = "westus"
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_storage_share" "testshare" {
  name = "sharename"

  resource_group_name  = "${azurerm_resource_group.test.name}"
  storage_account_name = "${azurerm_storage_account.test.name}"

  quota = 50
}
```

### Debug Output
```
Error: azurerm_resource_group.test: "name" may only contain alphanumeric characters, dash, underscores, parentheses and periods
Error: azurerm_storage_account.test: name can only consist of lowercase letters and numbers, and must be between 3 and 24 characters long
```

### Expected Behavior
Not sure.

### Actual Behavior
Could not deploy.

### Steps to Reproduce
1. `terraform apply`

### References
#493

**Comment by tombuildsstuff** [comment]
Hey @joshgav 

Thanks for opening this issue :)

> In the example for storage_share and storage_table a %d and %s are included in the names. The provider rejects this as an invalid name.
> To address this I submitted #493, are there better options?

Taking a look at the code both `azurerm_storage_share` and `azurerm_storage_table` have validation which catches this (which is the preferred option, since it's caught before any changes are made to Azure inside the `terraform plan`) - so I think in this case we can just update the documentation example as-in #493

Thanks!

**Comment by tombuildsstuff** [comment]
Closing since this has been fixed in #493
